### DISCUSSION TITLE
Arrow keys work incorrectly when typing song titles

### DISCUSSION THREAD

**Discussion by Cai1Hsu** [ppy/osu#20233] [Q&A]
### Type

Game behaviour

### Bug description

I was about to fix a typo when typing a song title, and found that the arrow keys work incorrectly. The bug still exists after i tried to make focus on the input box.

### Steps to reproduce:
1. launch osu! and go to the song select interface
2. type something.
3. press arrow key(left/right) to move the cursor 

### Expected result:
the cursor moves forward or backward.
### Actual result:
the selected song changed.

### Screenshots or videos

https://user-images.githubusercontent.com/47081443/189514354-43152213-a755-4f6b-9954-2a8eb76121ba.mp4

### Version

2022.910.0

### Logs

[performance.log](https://github.com/ppy/osu/files/9541849/performance.log)
[network.log](https://github.com/ppy/osu/files/9541850/network.log)
[database.log](https://github.com/ppy/osu/files/9541851/database.log)
[runtime.log](https://github.com/ppy/osu/files/9541852/runtime.log)
[updater.log](https://github.com/ppy/osu/files/9541853/updater.log)
[performance-update.log](https://github.com/ppy/osu/files/9541854/performance-update.log)
[performance-input.log](https://github.com/ppy/osu/files/9541855/performance-input.log)
[performance-draw.log](https://github.com/ppy/osu/files/9541856/performance-draw.log)
[performance-audio.log](https://github.com/ppy/osu/files/9541857/performance-audio.log)

**Comment by peppy** [comment]
This is intentional, to allow navigating song select.

Open to any suggestion you have.

**Reply by ghost** [reply]
Even if I change the relevant keybinds ("Select previous group" and "Select next group"), textbox navigation via arrow keys still don't work.

Another idea is to allow the textbox to be unselected, and pause song carousel logic when the textbox is selected.

**Reply by peppy** [reply]
> Even if I change the relevant keybinds ("Select previous group" and "Select next group"), textbox navigation via arrow keys still don't work.

That's definitely a weird edge case. Might be a hard one to resolve, I had a quick attempt but can't see what's eating the key presses (might be due to `PlatformAction` having a higher/lower priority at some internal level).

**Reply by frenzibyte** [reply]
Would imagine it's because of song select's use of `SeekLimitedSearchTextBox`, which straight up disables left/right keys:

https://github.com/ppy/osu/blob/f8830c6850128456266c82de83273204f8b74ac0/osu.Game/Graphics/UserInterface/SeekLimitedSearchTextBox.cs#L11-L14

**Reply by peppy** [reply]
Right. Seems that's actually required for things to work for now.

**Reply by tooruu** [reply]
I never use arrow keys to select beatmaps, instead I click on them with mouse/tablet.
But I always use my keyboard to search beatmaps (yes, really).

### My suggestion
After search box input event, introduce, let's say, a 5 second window, during which the arrow keys will move the cursor, rather than performing the action they are bound to.

### Alternative suggestion
If something is bound to either left or right arrow key, show a toggle on the song select screen that makes arrow keys move the cursor when it's switched on.

**Reply by Stoppedpuma** [reply]
Maybe holding control would work well? Similarly to how vim handles ^D and ^U to use alternate functions.
